alchemist/alchemist-implementationbase/src/main/kotlin/it/unibo/alchemist/model/implementations/geometry/IntervalsUtils.kt

```diff
@@ -11,14 +11,22 @@ package it.unibo.alchemist.model.implementations.geometry
 
 import it.unibo.alchemist.model.implementations.positions.Euclidean2DPosition
 import it.unibo.alchemist.model.interfaces.geometry.euclidean.twod.Segment2D
+import org.danilopianini.lang.MathUtils.fuzzyEquals
 import kotlin.math.max
 import kotlin.math.min
 
 /**
- * Checks if a value lies between two values (included) provided in any order.
+ * Checks if a double value lies between two values (included) provided in any order.
  */
 fun Double.liesBetween(v1: Double, v2: Double) = this >= min(v1, v2) && this <= max(v1, v2)
 
+/**
+ * Checks if a double value lies between two values (included) provided in any order with
+ * some tolerance.
+ */
+fun Double.fuzzyLiesBetween(v1: Double, v2: Double) =
+    liesBetween(v1, v2) || fuzzyEquals(this, v1) || fuzzyEquals(this, v2)
```

Comment:
Useless. You can use built-in Kotlin ranges:
x in v1..v2

Reply:
The convenience of `liesBetween` and `fuzzyLiesBetween` is that you can provide the two values in any order. Anyway, I inlined the latter since it was used only once, and replaced the former with Kotlin built-in ranges where possible.